source/slang/slang-check-expr.cpp

```diff
@@ -1553,7 +1557,11 @@ namespace Slang
         else
         {
             // Error
-            UNREACHABLE_RETURN(nullptr);
+            expr->type = astBuilder->getErrorType();
```

Comment:
@saipraveenb25 We always need to assign a type to an expr after checking it to prevent crashing the language server. If an expr fails checking, we assign ErrorType to it.